alert-city/server
# Feat: Reset Password Component

## TL;DR
- Optimized error exception and error filter
- Optimized Zod validation logic
- Optimized sendVerificationEmail and resetPassword methods


## Type of Change


## How Has This Been Tested?
1. open localhost:3000 ( or login into the system)
2. click forgot password
3. input username and click get code button
4. input verification code, new password, confirm new password and then click reset password
5. see the result 


## Screenshots:
Jest:
<img width="992" alt="image" src="https://github.com/user-attachments/assets/401bb28c-ec7b-4a2d-8fbc-869bb3d65737">


## Checklist


## Files for Review

1. validation/schemas/login/login.schemas.ts
2. validation/schemas/reset-password/reset-password.scheme.ts
3. validation/schemas/user/user.schemas.ts
4. user-utiles.service.ts
5. user.password.service.ts

## Resolve
Resolve CP-13
